node/src/main/scala/coop/rchain/node/proposescript/ValidatorProposeInfo.scala

```diff
@@ -0,0 +1,192 @@
+package coop.rchain.node.proposescript
+
+import cats.effect.Sync
+import cats.syntax.all._
+import coop.rchain.casper.protocol.{
+  BlockInfo,
+  BlockQuery,
+  BlocksQuery,
+  DeployData,
+  LightBlockInfo,
+  ProposeQuery,
+  ProposeResultQuery
+}
+import coop.rchain.casper.protocol.deploy.v1.DeployServiceV1GrpcMonix
+import coop.rchain.casper.protocol.propose.v1.ProposeServiceV1GrpcMonix
+import coop.rchain.crypto.PrivateKey
+import coop.rchain.crypto.codec.Base16
+import coop.rchain.crypto.signatures.{Secp256k1, Signed}
+import coop.rchain.monix.Monixable
+import coop.rchain.shared.Log
+import io.grpc.{ManagedChannel, ManagedChannelBuilder}
+import monix.eval.Task
+import coop.rchain.shared.syntax._
+
+final case class ProposeInfo(
+    deployContract: String,
+    phloLimit: Long,
+    phloPrice: Long,
+    deployPrivateKey: String
+)
+final case class ValidatorProposeInfo(
+    host: String,
+    grpcPort: Int,
+    httpPort: Int,
+    proposeInfo: ProposeInfo
+) {
+  private val ProposeSuccessRegex      = "Success! Block ([0-9a-f]+) created and added.".r
+  private val ProposeAsyncSuccessRegex = "Propose started \\(seqNum ([0-9]+)\\)".r
+
+  private val channel: ManagedChannel =
+    ManagedChannelBuilder
+      .forAddress(host, grpcPort)
+      .usePlaintext()
+      .build
```

Comment:
Closing the channel is not done now.

Let's make this whole class something like `ValidatorGrpcClient[F[_]: ...]` which accepts a channel. This whole class can be parameterized by `F`.

Channel should be wrapped in Resource so it can be closed gracefully.